lib/service/connect.go

```diff
@@ -792,8 +792,8 @@ func (process *TeleportProcess) newClient(authServers []utils.NetAddr, identity
 	log.Debugf("Attempting to connect to Auth Server directly.")
 	_, err = directClient.GetLocalClusterName()
 	if err != nil {
-		// Only attempt to connect through the proxy for nodes.
-		if identity.ID.Role != teleport.RoleNode {
+		// Don't attempt to connect through a tunnel as a proxy or auth server.
+		if identity.ID.Role == teleport.RoleAuth || identity.ID.Role == teleport.RoleProxy {
 			return nil, trace.Unwrap(err)
```

Comment:
I've never seen the error being actually unwrapped on return and just wanted to make sure it is intentional.

Reply:
Yeah, this looks like a typo.
If it's not, there should've been a comment explaining it.
Changed to `trace.Wrap`